src/test/java/com/exasol/adapter/document/DocumentAdapterIT.java

```diff
@@ -0,0 +1,109 @@
+    private static ExasolTestSetup testSetup;
+    private static ExasolObjectFactory exasolObjectFactory;
+    private static ConnectionDefinition nullConnection;
+    private static AdapterScript adapterScript;
+
+    @BeforeAll
+    static void beforeAll() throws SQLException, BucketAccessException, TimeoutException, FileNotFoundException {
+        testSetup = new ExasolTestcontainerTestSetup();
+        final UdfTestSetup udfTestSetup = new UdfTestSetup(testSetup);
+        exasolObjectFactory = new ExasolObjectFactory(testSetup.createConnection(),
+                ExasolObjectConfiguration.builder().withJvmOptions(udfTestSetup.getJvmOptions()).build());
+        buildMockAdapter();
+        final ExasolSchema adapterSchema = exasolObjectFactory.createSchema("ADAPTER");
+        testSetup.getDefaultBucket().uploadFile(Path.of("test-project", "mock-project", "target", MOCK_ADAPTER_JAR),
+                MOCK_ADAPTER_JAR);
+        adapterScript = adapterSchema.createAdapterScriptBuilder("FILES_ADAPTER")
+                .bucketFsContent("com.exasol.adapter.RequestDispatcher",
+                        "/buckets/bfsdefault/default/" + MOCK_ADAPTER_JAR)
+                .language(AdapterScript.Language.JAVA).build();
+        nullConnection = exasolObjectFactory.createConnectionDefinition("NULL_CONNECTION", "nowhere", "user", "pass");
+        createUdf(adapterSchema);
+    }
+
+    @AfterAll
+    static void afterAll() throws Exception {
+        testSetup.close();
+    }
+
+    private static void createUdf(final ExasolSchema adapterSchema) {
+        adapterSchema.createUdfBuilder("IMPORT_FROM_FIXED_DATA_ADAPTER").language(UdfScript.Language.JAVA)
+                .inputType(UdfScript.InputType.SET).parameter(PARAMETER_DOCUMENT_FETCHER, "VARCHAR(2000000)")
+                .parameter(PARAMETER_SCHEMA_MAPPING_REQUEST, "VARCHAR(2000000)")
+                .parameter(PARAMETER_CONNECTION_NAME, "VARCHAR(500)").emits()
+                .bucketFsContent(UdfEntryPoint.class.getName(), "/buckets/bfsdefault/default/" + MOCK_ADAPTER_JAR)
+                .build();
+    }
+
+    private static void buildMockAdapter() {
+        try {
+            LOGGER.info("Building mock-project");
+            final Verifier mvnRunner = new Verifier(Path.of("test-project", "aggregator").toAbsolutePath().toString());
+            mvnRunner.setSystemProperty("skipTests", "true");
+            mvnRunner.setSystemProperty("maven.test.skip", "true");
+            mvnRunner.setSystemProperty("ossindex.skip", "true");
+            mvnRunner.setSystemProperty("maven.javadoc.skip", "true");
+            mvnRunner.setSystemProperty("lombok.delombok.skip", "true");
+            mvnRunner.addCliOption("-PalternateTargetDir");
+            mvnRunner.executeGoal("package");
+            LOGGER.info("Done building mock-project");
+        } catch (final VerificationException exception) {
+            throw new IllegalStateException(ExaError.messageBuilder("E-VSD-76")
```

Comment:
Is it necessary to use an error ID in the integration test? The end user should never see this unless they run the tests themselves.

Reply:
Yes but we do it still. Is not only for end users but also for us. For example, it makes debugging easier if you know which error message you got.